player/command.c

```diff
@@ -515,28 +515,34 @@ static int mp_property_file_size(void *ctx, struct m_property *prop,
     return m_property_int64_ro(action, arg, size);
 }
 
-static int mp_property_media_title(void *ctx, struct m_property *prop,
-                                   int action, void *arg)
+static const char *find_non_filename_media_title(MPContext *mpctx)
 {
-    MPContext *mpctx = ctx;
-    char *name = NULL;
-    if (mpctx->opts->media_title)
-        name = mpctx->opts->media_title;
+    const char *name = mpctx->opts->media_title;
     if (name && name[0])
-        return m_property_strdup_ro(action, arg, name);
+        return name;
     if (mpctx->demuxer) {
         name = mp_tags_get_str(mpctx->demuxer->metadata, "service_name");
         if (name && name[0])
-            return m_property_strdup_ro(action, arg, name);
+            return name;
         name = mp_tags_get_str(mpctx->demuxer->metadata, "title");
         if (name && name[0])
-            return m_property_strdup_ro(action, arg, name);
+            return name;
         name = mp_tags_get_str(mpctx->demuxer->metadata, "icy-title");
         if (name && name[0])
-            return m_property_strdup_ro(action, arg, name);
+            return name;
     }
-    if (mpctx->playing && mpctx->playing->title)
-        return m_property_strdup_ro(action, arg, mpctx->playing->title);
+    if (mpctx->playing)
```

Comment:
Shouldn't this still be `if (mpctx->playing && mpctx->playing->title)` ?

Reply:
If it's not present it will return NULL, just like if the check fails.
But I guess that subtlety can be easily missed in case that ever gets extended, I'll change it.